### PR TITLE
Updates for the new Engine API

### PR DESCRIPTION
**Pull request by stubailo**
Todos:

P0 (sashko can not do)

- [x] Fill out API section @glasser 
- [x] Update version numbers everywhere @glasser 
- [x] Update places that refer to updating your engine config to make them easy to use with new API @glasser 

P1 (sashko can theoretically do)

- [x] Add migration guide from previous Engine version (`1.0-migration.html`) @glasser 
- [ ] Update "upgrade from optics" page @glasser
- [x] Update "setup troubleshooting" page and maybe delete the whole thing @glasser

### CODE REVIEW

**Comment by glasser** [comment]
Pages I haven't touched yet:

- `proto-doc.md` (I've changed the documentation of individual fields but not the overall confusingness)
- `standalone-proxy` (seems to be unlinked right now?)
- `setup-*` other than Node (going to merge many of these into one, and document ApolloEngineLauncher

There are also a couple final things that need to be updated once 1.0.1 is actually released.

File: source/caching.md [hunk 1]
```diff
@@ -38,11 +38,14 @@ app.use('/graphql', bodyParser.json(), graphqlExpress({
   cacheControl: true
 }));
 ```
+
+(Apollo Server includes built-in support for Apollo Cache Control from version 1.2.0 onwards. If you are using `express-graphql`, we recommend you switch to Apollo Server to use caching. Both `express-graphql` and Apollo Server are based on the [`graphql-js`](https://github.com/graphql/graphql-js) reference implementation, and switching should only require changing a few lines of code. We're working with the community to add support for Apollo Cache Control to non-Node GraphQL server libraries. [Contact us](mailto:support@apollographql.com) if you are interested in joining the community to work on support for `express-graphql` or non-Node GraphQL servers.)
```
Comment (stubailo) [inline source/caching.md]: I don't like having a whole paragraph in parentheses.

> Maybe try blockquote?

File: source/setup-standalone.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
+---
+title: Set up standalone Engine with any GraphQL server
+sidebar_title: Standalone Engine for any GraphQL server
+description: Get Engine running with your Ruby, Java, Scala, or Elixir server.
+---
+
+The easiest way to get Engine running is our [direct integration with Node.js web frameworks and Apollo Server](./setup-node.html).  But it takes only a little more setup to get Engine working with any GraphQL server.
```
Comment (stubailo) [inline source/setup-standalone.md]: Maybe:

> But it takes only a little more setup to get Engine working with any GraphQL API by running it as a proxy that processes requests before they hit your server.

File: source/setup-standalone.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+To get started with Engine, you will need to:
+1. Instrument your GraphQL server to enable the Apollo Tracing extension.
```
Comment (stubailo) [inline source/setup-standalone.md]: > Instrument your GraphQL server to add Apollo Tracing extension data to its responses.

File: source/setup-standalone.md (hunk 2 above)
Comment (glasser) [reply source/setup-standalone.md]: done

File: source/setup-standalone.md (hunk 3 above)
Comment (glasser) [reply source/setup-standalone.md]: done

File: source/caching.md (hunk 1 above)
Comment (glasser) [reply source/caching.md]: Eh just taking out of parens seems fine to me.

**Pull request merged**